EdgeReachTech/adgereachtech-web-bn
# Ci bra

### What does this PR do?
configuration of circle ci, docker file creation, adding testing module, configure jest and babel jest and minor change in exixting modules

### Description of Task to be completed?
i have created ci/cd :circle ci to automate build , development and test, implemented test module: unit test and for logi and register and integration test for both to test communication btn em using supertest

### How should this be manually tested?
to test circle ci you have to make push to our repo to triger it, to test docker file you have to run: docker create -p 5000:5000 :anyname you wish to assign to your image" and image will be created and you can run its conntainer and open browser for specified port you will see that application message. to test test module you have to run npm test to run all test all to run npm run test:unit for unit test and npm run test:integration to test integration

#### N/A

### Any background context you want to provide?
there is configuration of jest in project repo and babel to make test work, and i circle ci is working but testing is running unstopped because jest is not stopping due,  after test there is open handle, the running of app, and to stop it i think we have to set diffrent environment for our app and configure the app not not start when we are doing test. and i have removed packagelock jason in git ignore cause circle ci cant work cause it needs it

#### N/A

### Screenshots (if appropriate)

#### N/A

### Questions:
i think we have to make environment variables: for development, test and production

#### N/A
#
# On branch ci-bra
# Changes to be committed:
#	modified:   babel.config.js
#	new file:   jest.config.js
#	modified:   package.json
#	modified:   src/app.ts
#	modified:   src/controllers/userControllers.ts
#	modified:   src/services/userServices.ts
#	modified:   src/tests/integration/userIntegrationTest.test.js
#	renamed:    src/tests/unit/userTest.test.js -> src/tests/unit/userTestLogin.test.js
#	new file:   src/tests/unit/userTestRegister.test.js
#
